source/1_engine/rex_engine/include/rex_engine/memory/memory_manager.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include "rex_engine/types.h"
+#include "rex_std_extra/memory/memory_size.h"
+#include "rex_std_extra/utilities/high_water_mark.h"
+
+namespace rex
+{
+  class MemoryManager
+  {
+  public:
+    MemoryManager();
+
+    void initialize(rsl::memory_size maxMemUsage);
+
+    void track_alloc(u64 size);
+    void track_dealloc(u64 size);
+
+  private:
+    rsl::high_water_mark<s64> m_mem_usage; // current memory usage
```

Comment:
I would create a using for this to make the name of this object a little more clear. I have no idea type of object this is without looking it up.

To me a water mark is something that is put over an image, this has nothing to do with it.

Reply:
then a using won't make a different, you're suggesting a type name change, which is fine, but I don't know what else to call it.
it's a high water mark, like people do with rivers to track how high up it has come

Reply:
A high water mark is a point that represents the maximum rise of a body of water over land.
https://en.wikipedia.org/wiki/High_water_mark